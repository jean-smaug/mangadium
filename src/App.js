import React, { Component } from "react";
import { observer, inject } from "mobx-react";

import List from "./modules/List";
import Detail from "./modules/Detail";
import Search from "./modules/Search";
import Loader from "./components/Loader";

@inject("app")
@observer
class App extends Component {
  render() {
    return (
      <div>
        {this.props.app.isLoading && <Loader />}
        <Search />
        <List />
        <Detail />
      </div>
    );
  }
}

export default App;
