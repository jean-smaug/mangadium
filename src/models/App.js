import { types } from "mobx-state-tree";

const App = types
  .model({
    isLoading: types.optional(types.boolean, false)
  })
  .actions(self => ({
    toggleLoadingStatus() {
      self.isLoading = !self.isLoading;
    }
  }));

export default App;
