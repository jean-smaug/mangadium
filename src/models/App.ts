import { types } from 'mobx-state-tree'

export interface AppInterface{
  isLoading: boolean,
  toggleLoadingStatus: () => void
}

export const App = types
  .model({
    isLoading: types.optional(types.boolean, false)
  })
  .actions(self => ({
    toggleLoadingStatus () {
      self.isLoading = !self.isLoading
    }
  }))
