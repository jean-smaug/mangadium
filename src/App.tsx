import * as React from 'react'
import { observer, inject } from 'mobx-react'

import { AppInterface } from './models/App'
import Loader from './components/Loader'

import Search from './modules/Search'
import List from './modules/List'
import Detail from './modules/Detail'

@inject('app')
@observer
class App extends React.Component<AppInterface, {}> {
  render () {
    return (
      <div>
        hello
        {this.props.app.isLoading && <Loader />}
        <Search />
        <div style={{ marginTop: '70px' }}>
          <List />
          <Detail />
        </div>
      </div>
    )
  }
}

export default App
