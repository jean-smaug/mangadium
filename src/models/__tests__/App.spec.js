import App from '../App'

describe('<=== Manga ===>', () => {
  let app = null

  beforeEach(() => {
    app = App.create()
  })

  it('should create store', () => {
    expect(app).toMatchSnapshot()
  })

  it('should toggle loading status', () => {
    expect(app.isLoading).toBeFalsy()
    app.toggleLoadingStatus()
    expect(app.isLoading).toBeTruthy()
  })
})
