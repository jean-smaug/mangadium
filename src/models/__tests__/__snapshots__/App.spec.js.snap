// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<=== Manga ===> should create store 1`] = `
Object {
  "isLoading": false,
}
`;
